### PR TITLE
Feat: add get all tasks function

### PR DESCRIPTION
**Pull request by qwganker**

### CODE REVIEW

File: task/task.go [hunk 1]
```diff
@@ -653,6 +654,22 @@ func (m *taskManager) ClearTask() {
 	}
 }
 
+// GetAllTasks get all tasks
+func (m *taskManager) GetAllTasks() []*Task {
+	m.taskLock.RLock()
+
+	var l []*Task
+
+	for _, tasker := range m.adminTaskList {
```
Comment (flycash) [inline task/task.go:663]: why do you use reflection here?

File: task/task_test.go [hunk 2]
```diff
@@ -206,3 +206,28 @@ func wait(wg *sync.WaitGroup) chan bool {
...
+	tk2 := NewTask("task2", "0/40 * * * * *", func(ctx context.Context) error {
+		return nil
+	})
+
+	m.AddTask("task1", tk)
+	m.AddTask("task2", tk2)
+
+	l := m.GetAllTasks()
+	taskNum := 0
+	for _, t := range l {
```
Comment (flycash) [inline task/task_test.go:227]: use assertion in tests, and do not use any printing function, including `fmt.Println`

**Review by flycash — CHANGES_REQUESTED** [review]

**Comment by qwganker** [comment]
@flycash 

pls, review again,

**Comment by qwganker** [comment]
> I mean, why you use reflection in your code? you want to make a copy? but why we need a copy?

ok, now, i use another way to implment it that add GetTask() function into Tasker{} interface

**Comment by qwganker** [comment]
`DeepSource`  scan rule is changed?

File: task/task.go [hunk 3]
```diff
@@ -103,6 +103,7 @@ type TaskFunc func(ctx context.Context) error
 
 // Tasker task interface
 type Tasker interface {
+	GetTask(ctx context.Context) *Task
```
Comment (flycash) [inline task/task.go:106]: don't need this function

File: task/task.go [hunk 4]
```diff
@@ -653,6 +659,20 @@ func (m *taskManager) ClearTask() {
 	}
 }
 
+// GetAllTasks get all tasks
+func (m *taskManager) GetAllTasks() []*Task {
```
Comment (flycash) [inline task/task.go:663]: it should return the Tasker, and the users do the type assertion by themselves.
And the `taskManager` is a private structure, how do users call this function?

**Comment by qwganker** [comment]
@flycash  I updated the implement,  GetAllTasks() function will export to public, user  can use it like this
```golang
import beego "github.com/beego/beego/v2/task"

...

beego.AddTask("test1", t1)
beego.AddTask("test2", t2)

...

tasks := beego.GetAllTasks()
for _, task := range tasks {
  rv := reflect.ValueOf(task)
  t := rv.Interface().(*Task)
  fmt.Println(t)
}



```

File: task/task.go [hunk 5]
```diff
@@ -653,6 +658,20 @@ func (m *taskManager) ClearTask() {
 	}
 }
 
+// GetAllTasks get all tasks
+func (m *taskManager) GetAllTasks() []Tasker {
+	m.taskLock.RLock()
+
+	var l []Tasker
```
Comment (flycash) [inline task/task.go:665]:
```
l := make([]Tasker, 0, len(m.adminTaskList))
```
We can specify the capacity to avoid the memory allocation

**Review by flycash — CHANGES_REQUESTED** [review]

**Comment by flycash** [comment]
And do remember add one change log into `.CHANGELOG` file like this:
image

**Comment by qwganker** [comment]
@flycash  I has been updated, pls review again

**Review by flycash — APPROVED** [review]

**Pull request merged**